server/planning/core/service.py

```diff
@@ -8,5 +8,17 @@
 PlanningResourceModelType = TypeVar("PlanningResourceModelType", bound=BasePlanningModel)
 
 
-class BasePlanningAsyncService(AsyncResourceService[Generic[PlanningResourceModelType]]):
-    pass
+class BasePlanningAsyncService(AsyncResourceService[PlanningResourceModelType]):
+    """Base async service to handle repetitive logic that is used throughout the planning project."""
+
+    async def on_create(self, docs: List[PlanningResourceModelType]) -> None:
+        """
+        Sets `original_creator` and  `version_creator` by default to avoid repeating this everywhere
+        """
+        await super().on_create(docs)
+
+        user_id = get_user_id()
+        if user_id:
+            for doc in docs:
+                doc.original_creator = user_id
+                doc.version_creator = user_id
```

Comment:
I added an `on_create` to the Base class in this [PR](https://github.com/superdesk/superdesk-planning/pull/2172) which is merged but I got the user first then the ID. 

When you merge async into this we can keep yours since it's only the ID needed.